### ISSUE TITLE
Memory leak with text and FemtoVG renderer

### ISSUE THREAD

**Issue by tronical** [slint-ui/slint#2768]
### Discussed in https://github.com/slint-ui/slint/discussions/2750

<div type='discussions-op-text'>

<sup>Originally posted by **pcevent** May 17, 2023</sup>
Hello!
I made a small demo (using a timer to continuously increment a value on the interface). During operation, the memory is constantly increasing. Is it a bug or a cache caused? If it is caused by caching, how should it be turned off?

appwindow.slint
```
export component AppWindow inherits Window {
    property<int> counter: 42;
    callback inc();
    inc() => {
        counter += 1;
    }
    Text {
        text: "Counter: \{root.counter}";
    }
    width: 200px;
    height: 100px;
}
```

main.rust
```rust
slint::include_modules!();

use slint::Timer;

fn main() -> Result<(), slint::PlatformError> {
    let ui = AppWindow::new()?;

    let timer = Timer::default();
    let uh = ui.as_weak();
    timer.start(slint::TimerMode::Repeated, std::time::Duration::from_millis(20), move || {
        let ui = uh.unwrap();
        ui.invoke_inc();
    });

    ui.run()
}
```</div>

**Comment by tronical** [comment]
We've started work on trying to integrate cosmic-text for shaping. If all goes well that would allow us to avoid this particularly "leaky" code path in femtovg.

**Comment by puli1027** [comment]
Is this problem still unresolved?

**Comment by tronical** [comment]
Yes, it is unfortunately still a problem. I've started prototyping with parley instead of cosmic-text, but it's not anywhere near production ready-ness.

**Comment by blquinn** [comment]
I noticed quite a large memory leak in an app I'm toying around with https://github.com/blquinn/trixie

If you enable the femtovg backend, then try resizing the window a bunch, it'll pretty quickly eat up a gig of memory.

Not sure if it's this same leak though.

**Comment by tronical** [comment]
> I noticed quite a large memory leak in an app I'm toying around with https://github.com/blquinn/trixie
> 
> If you enable the femtovg backend, then try resizing the window a bunch, it'll pretty quickly eat up a gig of memory.
> 
> Not sure if it's this same leak though.

Memory use increase on window resizing is typically a symptom that appears on Windows with certain OpenGL drivers. Most likely it is not related to this issue, but something we’re going to address in one of the next releases when we turn off the OpenGL default on Windows. Already today you can give this a try by activating the `renderer-femtovg-wgpu` feature.

**Comment by tronical** [comment]
This is fixed in the master branch as of 0ce0fd80dde5656ae63690d2085f1dcc24054261 AFAICS. We don't use FemtoVG's shaper anymore, which had this word cache. The test in #9280 is also very useful for measuring this.
